### DISCUSSION TITLE
Suricata Container Fails on Forward Node

### DISCUSSION THREAD

**Discussion by cvfdblaze624** [Security-Onion-Solutions/securityonion#13533] [Unsupported Versions]
Distributed setup with master, search, forward node x1.  Recent update to 2.4.90 via SOUP and everything working for several days after, yesterday noticed Suricata wasn't running on forward node.  Have tried reboot, so-suricata-restart, pulled fresh docker images and rules on master and ran a high state to push changes.  Looking at the output of /opt/so/log/suricata/suricata.log on the forward node I see the following when trying to start or restart suricata.  

[11 - Suricata-Main] 2024-08-21 21:22:23 Error: mpm-ac: StateQueue behaving weirdly. Please file a bug report on this

I can't seem to find any information on this specific bug when searching so not sure what to try next.  I had some rules manually added to global.sls to threshold prior to the switch to detection method and I am getting a rule missmatch error on the detection tab so not sure if that has anything to do with my current issue or not.   Looking at the full Suricata log it seems to be loading all the rules without error so I don't think they are related.

**Comment by bushcraftbuddy** [comment]
I also am having this issue with a non-prod deployment. (no custom Rules at all) I have uninstalled and reinstalled the forward nodes with no change, so I assumed it is an issue with the manager node, but I have no custom configuration other than to enable Suricata. I don't want to reinstall my manager, but I will if I can't find a fix within the next couple weeks.

**Comment by TOoSmOotH** [comment]
Try disabling all your rules and try again. https://redmine.openinfosecfoundation.org/issues/2251

According to that bug its a file rule that is causing it possibly.

**Reply by bushcraftbuddy** [reply]
I disabled everything in the detections tab and it's back to working. I will dig into which rule caused this issue so we can find the exact issue

**Reply by tonyzuqhd** [reply]
how to disabled ?

**Reply by tonyzuqhd** [reply]
Bulk action:disable.   is ths ? I try,and wait.

**Reply by DrOnionMaster** [reply]
> Bulk action:disable. is ths ? I try,and wait.

Yes this is the fix. I can confirm I had the same problem and disabling all Suricata rules > restarting the node, worked.

**Comment by tonyzuqhd** [comment]
I have all the same problems that mentions above.And with the same situations:upgrade form a older version ,enable lots of rules in detections.So,wait for a answer.Thanks all.

**Reply by bushcraftbuddy** [reply]
what happens is when you enable all the rules you will have issues with matching rules from different rulesets. the ETOPEN ruleset has certain rules disabled for this reason. you can enable them on an as-needed basis instead of everything at once. currently the fix is to either disable everything or reinstall to refresh. I assume you could possibly find a way to delete all the rules and then re-pull them to set them back correctly 

![image](https://github.com/user-attachments/assets/aa7fd1de-1216-4504-8af3-d2f0f42011a3)

if you look at this newer install. there's a large number of rules automatically disabled by ET.  you may try to disable everything but a certain Ruleset: ETOPEN, Core, etc. but I don't know exactly what rules are causing this issue.

Note that if Suricata is failing then ETOPEN is the ruleset you should disable first. that should fix it after a reboot. that does narrow down what rules could be causing the issue down to 51,796

**Comment by cvfdblaze624** [comment]
I have got it back up after disabling all ETOPEN rules and then re-enabling them.  Has only been back up a short time but before every time I would try to restart the service it would quickly fail again.  On the detection page I am now seeing a rule missmatch error on both ElasticAlert and Suricata.
